gotran/codegeneration/solvercomponents.py

```diff
@@ -696,20 +736,28 @@ def __init__(
                 )
                 continue
 
+            linearized_name = expr.name + "_linearized"
             linearized = self.add_intermediate(
-                expr.name + "_linearized", expr_diff, dependent=dependent
+                linearized_name, expr_diff, dependent=dependent
             )
 
+            need_zero_div_check = not fraction_numerator_is_nonzero(expr_diff)
+            if not need_zero_div_check:
+                debug("{} cannot be zero. Skipping zero division check".format(linearized_name))
+
+            RL_term = expr.sym / linearized * (sp.exp(linearized * dt) - 1)
+            if need_zero_div_check:
+                RL_term = Conditional(
+                    abs(linearized) > delta,
+                    RL_term,
+                    dt * expr.sym,
+                )
+
```

Comment:
More repetition (but I guess `gotran` contains quite a lot of repetetive code in general)